src/test/java/net/snowflake/ingest/streaming/internal/datatypes/IcebergNumericTypesIT.java

```diff
@@ -368,5 +379,48 @@ public void testDecimalAndQueries() throws Exception {
         Arrays.asList(new BigDecimal("-12.3"), new BigDecimal("-12.3"), null),
         "select COUNT({columnName}) from {tableName} where {columnName} = -12.3",
         Arrays.asList(2L));
+
+    List<Object> bigDecimals_9_4 = randomBigDecimal(200, 9, 4);
+    testIcebergIngestAndQuery(
+        "decimal(9, 4)", bigDecimals_9_4, "select {columnName} from {tableName}", bigDecimals_9_4);
+
+    List<Object> bigDecimals_18_9 = randomBigDecimal(200, 18, 9);
+    testIcebergIngestAndQuery(
+        "decimal(18, 9)",
+        bigDecimals_18_9,
+        "select {columnName} from {tableName}",
+        bigDecimals_18_9);
+
+    List<Object> bigDecimals_21_0 = randomBigDecimal(200, 21, 0);
+    testIcebergIngestAndQuery(
+        "decimal(21, 0)",
+        bigDecimals_21_0,
+        "select {columnName} from {tableName}",
+        bigDecimals_21_0);
+
+    List<Object> bigDecimals_38_10 = randomBigDecimal(200, 38, 10);
+    testIcebergIngestAndQuery(
+        "decimal(38, 10)",
+        bigDecimals_38_10,
+        "select {columnName} from {tableName}",
+        bigDecimals_38_10);
+  }
+
+  private static List<Object> randomBigDecimal(int count, int precision, int scale) {
+    List<Object> list = new ArrayList<>();
+    for (int i = 0; i < count; i++) {
+      int intPart = generator.nextInt(precision - scale + 1);
+      int floatPart = generator.nextInt(scale + 1);
+      if (intPart == 0 && floatPart == 0) {
+        list.add(null);
+        continue;
+      }
+      list.add(
+          new BigDecimal(
+              RandomStringUtils.randomNumeric(intPart)
```

Comment:
this makes it impossible to reproduce failures locally. Why do we need this randomNumeric when we already generated a random number for intpart and floatpart using the seeded RNG in lines 412 and 413?

Reply:
The length of decimal is greater than int/long.max, thus we need string generator for this. Switch to generator with consistent seed.